libwebthree/WebThree.cpp

```diff
@@ -56,13 +56,12 @@ WebThreeDirect::WebThreeDirect(std::string const& _clientVersion,
             m_ethereum.reset(new eth::Client(_params, (int)_params.networkID, &m_net, shared_ptr<GasPricer>(), _dbPath, _snapshotPath, _we));
         m_ethereum->startWorking();
 
-        string bp = DEV_QUOTED(ETH_BUILD_PLATFORM);
-        vector<string> bps;
-        boost::split(bps, bp, boost::is_any_of("/"));
-        bps[0] = bps[0].substr(0, 5);
-        bps[1] = bps[1].substr(0, 3);
-        bps.back() = bps.back().substr(0, 3);
-        m_ethereum->setExtraData(rlpList(0, string(dev::Version) + "++" + string(DEV_QUOTED(ETH_COMMIT_HASH)).substr(0, 4) + (ETH_CLEAN_REPO ? "-" : "*") + string(DEV_QUOTED(ETH_BUILD_TYPE)).substr(0, 1) + boost::join(bps, "/")));
+        const auto* buildinfo = eth_get_buildinfo();
+        m_ethereum->setExtraData(rlpList(0, string{buildinfo->project_version}.substr(0, 5) + "++" +
```

Comment:
We've lost `ETH_CLEAN_REPO` here? Do you know whether it worked before?

Reply:
These magic values for substr lengths look like they will break the first time some of the formats of build info strings changes.
Maybe we should have a function in buildinfo lib that produces a compact string that includes all info?

Also, extraData is 32 bytes I think, we could fit there more symbols if needed.

Reply:
Yes, we lost it. I believe it was working before.
The dirty part is included in `project_version`, because it looks like `1.4.0.dev0-N+commit.XXXXXXXX-dirty`. But here this "project version" is trimmed to 5 chars.
I can add the "dirty" information to `buildinfo` struct similarly as there is `git_commit_hash`. But I don't think this information is super important.

Reply:
I agree the extra data here is messy and fragile. Still `string{...}.substr(0, x)` pattern should always work. I would also get rid of RLP here.
I didn't wanted to touch it in this PR.